integration/spark-common/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchemaCommon.scala

```diff
@@ -393,13 +393,6 @@ class TableNewProcessor(cm: TableModel) {
     var allColumns: Seq[ColumnSchema] = Seq[ColumnSchema]()
     fieldChildren.foreach(fields => {
       fields.foreach(field => {
-        if (!useDictionaryEncoding &&
```

Comment:
Don't remove this check in normal flow.

Reply:
This was added to block double/date/decimal for complex type in case of dictionary exclude as now it is supported remove the same. this check is only applicable for complex type children